compiler/src/iree/compiler/Dialect/Stream/Analysis/ResourceUsage.cpp

```diff
@@ -548,6 +561,19 @@ class ValueResourceUsage : public AbstractResourceUsage<DFX::ValueElement> {
                 *this, value, DFX::Resolution::REQUIRED);
             getState() ^= valueUsage.getState();
           }
+
+          if (auto forOp = dyn_cast_or_null<scf::ForOp>(op->getParentOp())) {
+            auto value = Position::forValue(
+                forOp.getRegion().front().getArgument(operandIdx + 1));
```

Comment:
same here with operandIdx indexing if we could have something more structured/less magic

Reply:
Fixed and removed magic numbers